### PR TITLE
[Bug][DOE]: fix doe on es8

### PR DESCRIPTION
**Pull request by stalary**
## Problem Summary:

doris on es8 can not work, because type change. The use of type is no longer recommended in es7, and support for type has been removed from es8.
1. /_mapping not support include_type_name
2. /_search not support use type

## Further comments
1. Complete the documentation and recommend passing in the cluster version when create the table
2. Be passes type logic to add judgment
3. include_type_name is used only in es7

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM
but please add some detail in your PR comment.

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/MappingPhase.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public MappingPhase(EsRestClient client) {
 
     @Override
     public void preProcess(SearchContext context) {
-        if (context.version() != null && context.version().onOrAfter(EsMajorVersion.V_7_X)) {
+        if (context.version() != null && context.version().on(EsMajorVersion.V_7_X)) {
```
Comment (wuyunfeng) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/MappingPhase.java:48]: Remove this `prePrcess`

File: fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java [hunk 2]
```diff
@@ -250,7 +251,10 @@ private List<TScanRangeLocations> getShardLocations() throws UserException {
                 TEsScanRange esScanRange = new TEsScanRange();
                 esScanRange.setEsHosts(shardAllocations);
                 esScanRange.setIndex(shardRouting.get(0).getIndexName());
-                esScanRange.setType(table.getMappingType());
+                // 8.x type is cancelled
+                if (table.majorVersion == null || table.majorVersion.before(EsMajorVersion.V_8_X)) {
```
Comment (wuyunfeng) [inline fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java:255]: Don't rely on this, don't issue this parameter directly, BE will do the empty judgment processing

**Review by wuyunfeng — CHANGES_REQUESTED** [review]

**Comment by wuyunfeng** [comment]
@stalary  Thanks for your contribution.  I suggest remove the `include_type_name` branch, and do not pass `type` in ESScanNode,  BE check whether the `type` parameter is empty and choose a different URL.  And, Do not forget modify catalog logical for ES table when invoke `show create table DOE`

File: be/src/exec/es/es_scan_reader.cpp [hunk 3]
```diff
@@ -47,7 +47,11 @@ ESScanReader::ESScanReader(const std::string& target,
           _doc_value_mode(doc_value_mode) {
     _target = target;
     _index = props.at(KEY_INDEX);
-    _type = props.at(KEY_TYPE);
+    if (props.find(KEY_TYPE) != props.end()) {
+        _type = REQUEST_SEPARATOR + props.at(KEY_TYPE);
+    } else {
+        _type = "";
```
Comment (wuyunfeng) [inline be/src/exec/es/es_scan_reader.cpp:53]: remove this，default is empty

**Comment by stalary** [comment]
I have finished the modification @wuyunfeng

File: docs/en/docs/ecosystem/external-table/doris-on-es.md [hunk 4]
```diff
@@ -121,7 +121,7 @@ PROPERTIES (
 "hosts" = "http://192.168.0.1:8200,http://192.168.0.2:8200",
 "index" = "test",
 "type" = "doc",
-
+"version" = "6.8.23"
```
Comment (wuyunfeng) [inline docs/en/docs/ecosystem/external-table/doris-on-es.md:124]: remove this line

File: fe/fe-core/src/main/java/org/apache/doris/catalog/EsTable.java [hunk 5]
```diff
@@ -73,7 +73,7 @@ public class EsTable extends Table {
     private String indexName;
 
     // which type used for `indexName`, default to `_doc`
```
Comment (wuyunfeng) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/EsTable.java:75]:
```suggestion
    // which type used for `indexName`
```

File: fe/fe-core/src/main/java/org/apache/doris/catalog/EsTable.java [hunk 6]
```diff
@@ -263,8 +261,8 @@ private void validate(Map<String, String> properties) throws DdlException {
 
     public TTableDescriptor toThrift() {
         TEsTable tEsTable = new TEsTable();
-        TTableDescriptor tTableDescriptor = new TTableDescriptor(getId(), TTableType.ES_TABLE,
-                fullSchema.size(), 0, getName(), "");
+        TTableDescriptor tTableDescriptor = new TTableDescriptor(getId(), TTableType.ES_TABLE, fullSchema.size(), 0,
+                getName(), "");
```
Comment (wuyunfeng) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/EsTable.java:265]: getSignature method should process  mappingType is null

**Review by wuyunfeng — CHANGES_REQUESTED** [review]

**Review by wuyunfeng — APPROVED** [review]

**Pull request merged**